sds_data_manager/lambda_code/SDSCode/create_schema.py

```diff
@@ -17,18 +18,48 @@
 
 def lambda_handler(event, context):
     """Entry point to the create schema lambda."""
-    logger.info("Creating RDS tables")
-    logger.info(event)
+    logger.info(f"Event: {event}")
 
-    # NOTE: If we run this when trying to delete the stack, it fails because
-    #       the RDS isn't available, so skip it here.
-    if event.get("RequestType") != "Create":
-        logger.info("Skipping schema creation, only handling Create requests")
+    if event.get("RequestType") == "Delete":
+        logger.info(
+            "Skipping schema modification, only handling Create or Update requests"
+        )
         return
 
-    # Create tables
-    Base.metadata.create_all(db.get_engine())
-
     with db.Session() as session:
-        session.add_all(all_dependents)
-        session.commit()
+        if event.get("RequestType") == "Create":
+            logger.info("Creating RDS tables")
+            # Create tables
+            Base.metadata.create_all(db.get_engine())
+            session.add_all(all_dependents)
+
+        elif event.get("RequestType") == "Update":
+            for dependent in all_dependents:
+                existing_record = (
```

Comment:
Can we make a unique index on the database to avoid duplicates? It seems like we should be able to do a large insert/update of all of the items. If there are entries already this would just "update" that entry. Maybe we need to have a better index set up to handle this instead of iterating through items.